src/main/java/org/testng/IAlterTestName.java

```diff
@@ -0,0 +1,19 @@
+package org.testng;
+
+/**
+ * This interface lets you alter the test name of an {@link ITestResult} object.
+ * Sample : <br>
+ *
+ * <pre>
+ *     if (testResult instanceOf IAlterTestName) {
+ *         ((ITestResult) testResult).setTestName(newName);
+ *     }
+ *
+ * </pre>
+ */
+public interface IAlterTestName {
```

Comment:
Yes. That was what I was hinting at. But like I said I dont know if its a fair assumption. But when I searched on Github I found [this](https://github.com/joel-costigliola/assertj-swing/blob/master/assertj-swing-testng/src/test/java/org/assertj/swing/testng/listener/TestResultStub.java) wherein someone is implementing the `ITestResult` interface. So I guess this is a valid assumption.